classes/class-paysoncheckout-for-woocommerce-order-management.php

```diff
@@ -106,9 +105,10 @@ public function cancel_reservation( $order_id ) {
 		// Check if we where successfull.
 		if ( ! is_wp_error( $payson_order ) && 'canceled' == $payson_order['status'] ) {
 			// Add time stamp, used to prevent duplicate cancellations for the same order.
-			update_post_meta( $order_id, '_paysoncheckout_reservation_cancelled', current_time( 'mysql' ) );
+			$order->update_meta_data( '_paysoncheckout_reservation_cancelled', current_time( 'mysql' ) );
 			// Add Payson order status.
-			update_post_meta( $order_id, '_paysoncheckout_order_status', $payson_order['status'] );
+			$order->update_meta_data( $order_id, '_paysoncheckout_order_status', $payson_order['status'] );
```

Comment:
It seems like you've accidentally left `$order_id`. This should be removed accordingly:

```php
$order->update_meta_data( '_paysoncheckout_order_status', $payson_order['status'] );
```